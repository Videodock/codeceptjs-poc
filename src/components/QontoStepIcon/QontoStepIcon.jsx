import React from 'react';
import { makeStyles } from '@material-ui/styles';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './QontoStepIcon.styles';

const useStyles = makeStyles(styles);

export default function QontoStepIcon ({ active, completed }) {
  const classes               = useStyles();

  return (
    <div className={classes.root}>
      {active &&       
      <div
        className={classNames(classes.circle, {
          [classes.active]: active,
        })}
      />}
      { (!completed && !active) &&  <div className={classes.circle} />}
    </div>
  );
}

QontoStepIcon.propTypes = {
  active   : PropTypes.bool,
  completed: PropTypes.bool,
};
