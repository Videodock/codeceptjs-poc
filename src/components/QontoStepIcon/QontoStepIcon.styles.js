import { createStyles } from '@material-ui/styles';

export default ({ palette }) => createStyles({
  root: {
    color     : '#eaeaf0',
    display   : 'flex',
    height    : 22,
    alignItems: 'center',
  },
  circle: {
    width          : 5,
    height         : 5,
    marginTop      : 2.5,
    borderRadius   : '50%',
    backgroundColor: palette.common.borderGrey,
  },
  active: {
    backgroundColor: '#784af4',
  },
});
