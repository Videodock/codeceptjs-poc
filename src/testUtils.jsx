import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
