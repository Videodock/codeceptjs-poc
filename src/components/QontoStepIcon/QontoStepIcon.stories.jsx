import React from 'react';
import { storiesOf } from '@storybook/react';
import QontoStepIcon from './QontoStepIcon';

storiesOf('QontoStepIcon', module)
  .add('basic', () => <QontoStepIcon />);
