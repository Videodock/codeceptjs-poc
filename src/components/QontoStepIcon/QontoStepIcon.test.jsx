import React from 'react';
import { render } from '../../testUtils';
import QontoStepIcon from './QontoStepIcon';

describe('<QontoStepIcon />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<QontoStepIcon />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
