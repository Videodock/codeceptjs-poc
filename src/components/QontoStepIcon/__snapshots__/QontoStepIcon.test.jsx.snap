// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QontoStepIcon /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-circle-2"
  />
</div>
`;
